### PR TITLE
Send user updates to aurora

### LINKED ISSUE #4549: Undefined Northstar user attributes
Notice: Undefined index: und in dosomething_northstar_build_ns_user() (line 75 of /var/www/staging.beta.dosomething.org/releases/20150601213157/lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module).

Only send attributes if they exist, in this case, there was no mobile number set.

### LINKED ISSUE #4570: Forward user updates from Drupal to Northstar
When users update their profile on Drupal, ensure these updates are also reflected on Northstar.

Guzzle should be available to use soon: #4411

### PR DESCRIPTION
**Pull request by angaither**
On use profile saves, this sends the user array into northstar.
Also refactored user registration to use guzzle to send requests. 

Fixes #4549
Fixes #4570

### CODE REVIEW

File: lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module [hunk 1]
```diff
@@ -107,41 +148,39 @@ function _dosomething_northstar_build_http_client() {
...
-//       ));
-//   }
-
-// }
+class Northstar {
+
+  protected $client;
+
+  public function __construct() {
+    $this->client = $library;
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module]: How is `$library` defined? In line 163?

File: lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module [hunk 2]
```diff
@@ -107,41 +148,39 @@ function _dosomething_northstar_build_http_client() {
...
+    $base_url = NORTHSTAR_URL;
+    if (getenv('DS_ENVIRONMENT') === 'local') {
+      $base_url .=  ":" . NORTHSTAR_PORT;
+    }
+    $version = NORTHSTAR_VERSION;
+
+    $library = libraries_load('guzzle');
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module]: Need to ensure that `libraries_load` function returns TRUE before calling below.

Comment (angaither) [inline lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module]: updated!

**Comment by aaronschachter** [comment]
:+1:

**Pull request merged**